public/services/data_source_service.mock.ts

```diff
@@ -4,12 +4,18 @@
  */
 
 export class DataSourceServiceMock {
-  constructor() {}
+  private _isMDSEnabled = false;
+  constructor(isMDSEnabled?: boolean) {
+    this._isMDSEnabled = isMDSEnabled ?? false;
+  }
 
   getDataSourceQuery() {
```

Comment:
Shall we resolve an empty object for MDS not enabled?

Reply:
I don't think so. getDataSourceQuery is a function that would be called all time, although it would return empty string when MDS disabled.

Reply:
Just to keep consistent mock logic with the real world in case of some place set `isMDSEnabled` false but the `getDataSourceQuery` still return dataSourceId.

Reply:
I have updated getDataSourceQuery function in mock to make it consistent.